types/base.d.ts

```diff
@@ -120,5 +120,13 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 	context: any;
 }
 
+export interface Macro<T, I, E> {
+	(t: T, input: I, expected: E): void;
+	title? (providedTitle: string, input: I, expected: E): string;
+}
+export type Macros<T, I, E> = Macro<T, I, E> | Macro<T, I, E>[];
+
 export function test(name: string, run: ContextualTest): void;
 export function test(run: ContextualTest): void;
+export function test<I, E> (name: string, run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
+export function test<I, E> (run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
```

Comment:
> Support max two arguments, and mark them optional (?) - this means that the amount of arguments isn't checked; a macro with 2 arguments can be run without arguments, and the other way around

I think this would be the best solution. At least it would then typecheck the first two arguments. We should open an issue to improve this when https://github.com/Microsoft/TypeScript/issues/5453 is fixed.